pkg/registry/core/service/storage/storage.go

```diff
@@ -625,6 +625,12 @@ func patchAllocatedValues(after After, before Before) {
 		for i := range newSvc.Spec.Ports {
 			p := &newSvc.Spec.Ports[i]
 			if p.NodePort == 0 {
+				// skip setting NodePort in case AllocateLoadBalancerNodePorts is false
+				if newSvc.Spec.Type == api.ServiceTypeLoadBalancer &&
+					newSvc.Spec.AllocateLoadBalancerNodePorts != nil &&
+					!*newSvc.Spec.AllocateLoadBalancerNodePorts {
+					continue
+				}
```

Comment:
this will modify the behavior, a patch can not deallocate the nodePorts

Reply:
you have to remove this

Reply:
updated